src/coreComponents/physicsSolvers/multiphysics/LagrangianContactSolver.cpp

```diff
@@ -1040,6 +1108,173 @@ real64 LagrangianContactSolver::CalculateResidualNorm( DomainPartition const & d
   return globalResidualNorm[2];
 }
 
+void LagrangianContactSolver::CreatePreconditioner( DomainPartition const & domain )
+{
+  if( m_linearSolverParameters.get().preconditionerType == LinearSolverParameters::PreconditionerType::block )
+  {
+    // TODO: move among inputs (xml)
+    std::string const leadingBlockApproximation = "blockJacobi";
```

Comment:
I would rather keep this as a separate preconditioner, since I need it in _matrix_ form, not only as `M^-1*v` application. This is required to compute the Schur complement. But we can develop this point in a different PR.

Reply:
I see your point. Let's keep everything as is.

Reply:
Ok!